stubs/PyYAML/yaml/__init__.pyi

```diff
@@ -258,12 +261,46 @@ def safe_dump(
     tags=...,
     sort_keys: bool = ...,
 ) -> _Yaml: ...
-def add_implicit_resolver(tag, regexp, first=..., Loader=..., Dumper=...): ...
-def add_path_resolver(tag, path, kind=..., Loader=..., Dumper=...): ...
-def add_constructor(tag: _Str, constructor: Callable[[Loader, Node], Any], Loader: Loader = ...): ...
-def add_multi_constructor(tag_prefix, multi_constructor, Loader=...): ...
-def add_representer(data_type: Type[_T], representer: Callable[[_R, _T], Node], Dumper: Type[_R] = ...) -> None: ...
-def add_multi_representer(data_type: Type[_T], multi_representer: Callable[[_R, _T], Node], Dumper: Type[_R] = ...) -> None: ...
+def add_implicit_resolver(
+    tag: _Str,
+    regexp: Pattern[str],
+    first: Iterable[Any] | None = ...,
+    Loader: Type[BaseResolver] | None = ...,
+    Dumper: Type[BaseResolver] = ...,
+) -> None: ...
+def add_path_resolver(
+    tag: _Str,
+    path: Iterable[Any],
+    kind: Type[Any] | None = ...,
+    Loader: Type[BaseResolver] | None = ...,
+    Dumper: Type[BaseResolver] = ...,
+) -> None: ...
+@overload
+def add_constructor(
+    tag: _Str, constructor: Callable[[Loader | FullLoader | UnsafeLoader, Node], Any], Loader: None = ...
+) -> None: ...
+@overload
+def add_constructor(tag: _Str, constructor: Callable[[_Constructor, Node], Any], Loader: Type[_Constructor] = ...) -> None: ...
```

Comment:
The second overload shouldn't have a default value. The first overload covers the case where the default value is used.

```suggestion
def add_constructor(tag: _Str, constructor: Callable[[_Constructor, Node], Any], Loader: Type[_Constructor]) -> None: ...
```

Reply:
Makes sense, thanks for pointing out. Fixed all four cases in dae51a6.